### PR TITLE
stm32: update optee-os, u-boot and arm-trusted-firmware

### PR DESCRIPTION
**Pull request by thom24**
Update optee-os, u-boot and arm-trusted-firmware for stm32 target

### CODE REVIEW

File: package/boot/arm-trusted-firmware-stm32/patches/0001-Revert-feat-stm32mp1-fdts-remove-RTC-clock-configura.patch
Comment (hauke) [inline package/boot/arm-trusted-firmware-stm32/patches/0001-Revert-feat-stm32mp1-fdts-remove-RTC-clock-configura.patch:1]: Please run `make package/arm-trusted-firmware-stm32/refresh` to refresh the patch

File: package/boot/arm-trusted-firmware-stm32/Makefile [hunk 1]
```diff
@@ -7,10 +7,13 @@
...
-PKG_VERSION:=2.10
+PKG_VERSION:=2.12
 PKG_RELEASE:=1
 
-PKG_HASH:=88215a62291b9ba87da8e50b077741103cdc08fb6c9e1ebd34dfaace746d3201
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/ARM-software/arm-trusted-firmware.git
+PKG_SOURCE_VERSION:=v$(PKG_VERSION)
+PKG_MIRROR_HASH:=51b2022baa25df7fd8f2e6d2709c9351c14b17447cda64759a8a1d432f9d1c11
 PKG_MAINTAINER:=Thomas Richard <thomas.richard@bootlin.com>
```
Comment (hauke) [inline package/boot/arm-trusted-firmware-stm32/Makefile:17]: Why are you changing the URL?

Comment (1715173329) [reply package/boot/arm-trusted-firmware-stm32/Makefile:17]: Current URL `https://git.trustedfirmware.org/TF-A/trusted-firmware-a.git/snapshot` does not ship new tarballs.

**Comment by hauke** [comment]
This breaks the build. It is failing like this in our build bots"
```
recated-declarations -Wno-error=cpp -DIMAGE_   -mthumb -mno-unaligned-access -P -x assembler-with-cpp -D__LINKER__ -Wp,-MD,/builder/shared-workdir/build/build_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/trusted-
firmware-a-stm32mp135f-dk/trusted-firmware-a-2.12/build/stm32mp1/release/stm32mp1.ld.d -MT /builder/shared-workdir/build/build_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/trusted-firmware-a-stm32mp135f-dk/truste
d-firmware-a-2.12/build/stm32mp1/release/stm32mp1.ld -MP -o /builder/shared-workdir/build/build_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/trusted-firmware-a-stm32mp135f-dk/trusted-firmware-a-2.12/build/stm32mp
1/release/stm32mp1.ld plat/st/stm32mp1/stm32mp1.ld.S
gcc-10: error: unrecognized command-line option '-mthumb'
gcc-10: error: unrecognized command-line option '-mno-unaligned-access'
make[3]: *** [plat/st/common/common_rules.mk:57: /builder/shared-workdir/build/build_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/trusted-firmware-a-stm32mp135f-dk/trusted-firmware-a-2.12/build/stm32mp1/release/s
tm32mp1.ld] Error 1
make[3]: Leaving directory '/builder/shared-workdir/build/build_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/trusted-firmware-a-stm32mp135f-dk/trusted-firmware-a-2.12'
make[2]: *** [Makefile:69: /builder/shared-workdir/build/build_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/trusted-firmware-a-stm32mp135f-dk/trusted-firmware-a-2.12/.built] Error 2
make[2]: Leaving directory '/builder/shared-workdir/build/package/boot/arm-trusted-firmware-stm32'
time: package/boot/arm-trusted-firmware-stm32/stm32mp135f-dk/compile#0.84#0.82#1.97
    ERROR: package/boot/arm-trusted-firmware-stm32 failed to build (build variant: stm32mp135f-dk).
make[1]: *** [package/Makefile:185: package/boot/arm-trusted-firmware-stm32/compile] Error 1
make[1]: *** Waiting for unfinished jobs....
```

I think it is using the host gcc.

**Comment by thom24** [comment]
@hauke, it is weird.
On my side the cross gcc is used, so I cannot reproduce the error.
I identified the line, but it seems ok for me:
https://github.com/ARM-software/arm-trusted-firmware/blob/v2.12.0/make_helpers/build_macros.mk#L387

I don't understand why the build bot uses its host gcc.

**Pull request merged**